java-frontend/src/main/java/org/sonar/java/cfg/LiveVariables.java

```diff
@@ -32,44 +42,41 @@
 import org.sonar.plugins.java.api.tree.Tree;
 import org.sonar.plugins.java.api.tree.VariableTree;
 
-import javax.annotation.Nullable;
-import java.util.Collections;
-import java.util.Deque;
-import java.util.HashMap;
-import java.util.HashSet;
-import java.util.LinkedList;
-import java.util.List;
-import java.util.Map;
-import java.util.Set;
-
 public class
     LiveVariables {
 
-  private final CFG cfg;
-  private final Map<CFG.Block, Set<Symbol>> out = new HashMap<>();
+  protected final CFG cfg;
+  protected final Map<CFG.Block, Set<Symbol>> out = new HashMap<>();
+  protected final Map<Block, Set<Symbol>> in = new HashMap<>();
 
-  private LiveVariables(CFG cfg) {
+  protected LiveVariables(CFG cfg) {
     this.cfg = cfg;
   }
 
+  protected static LiveVariables create(CFG cfg) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Protected method 'create(...)' is never used. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUnusedProtectedMethod)